Rocket.ChatTests/Controllers/ConnectServerViewControllerSpec.swift

```diff
@@ -0,0 +1,23 @@
+//
+//  ConnectServerViewControllerSpec.swift
+//  Rocket.ChatTests
+//
+//  Created by Matheus Cardoso on 10/5/17.
+//  Copyright © 2017 Rocket.Chat. All rights reserved.
+//
+
+import XCTest
+@testable import Rocket_Chat
+
+class ConnectServerViewControllerSpec: XCTestCase {
+    func testNormalizeInputURL() {
+        let vc = ConnectServerViewController()
+
+        XCTAssertEqual(vc.normalizeInputURL("open.rocket.chat"), "https://open.rocket.chat", "will add https")
+        XCTAssertEqual(vc.normalizeInputURL("https://open.rocket.chat"), "https://open.rocket.chat", "will return correct url")
+        XCTAssertEqual(vc.normalizeInputURL("http://open.rocket.chat"), "https://open.rocket.chat", "will force https scheme")
+        XCTAssertNil(vc.normalizeInputURL("https://"), "will return nil when hostless")
+        XCTAssertNil(vc.normalizeInputURL(""), "will return nil when empty")
+    }
```

Comment:
👏

Reply:
I would also include a test here that verifies if the path keeps on the URL and some query parameters:

- `https://open.rocket.chat/chat`
- `open.rocket.chat/chat`
- `open.rocket.chat/chat/foo`
- `open.rocket.chat/chat/foo?test=1`
- `open.rocket.chat/chat/foo?test=1&test=2`

What do you think?

Reply:
They actually don't. I didn't realize that was the intended behavior.

Reply:
@cardoso Yes, there's some cases when you can use path and maybe some query URL? :-)

Reply:
@cardoso You can use this method that I created long time ago to generate the URL in the API abstraction...

https://github.com/RocketChat/Rocket.Chat.iOS/blob/2d1eea6f62310d274713b9568dd7a97181f5f23c/Rocket.Chat/Extensions/URLExtension.swift#L26-L34

Reply:
You can see that I even consider the port here ^^